Documentation/ConfigurationManagement.md

```diff
@@ -0,0 +1,97 @@
+# Configuration Management
+
+This documentation provides a comprehensive solution for integrating and managing configuration files in Open edX Android project.
+
+## Features
+- Parsing config.yaml files
+- Adding essential keys to `AndroidManifest.xml` (e.g. Microsoft keys)
+- Generating Android build config fields.
+- Generating config.json file to use the configuration fields at runtime.
+- Generating google-services.json with Firebase keys.
+
+Inside the `Config.kt`, parsing and populating relevant keys and classes are done, e.g. `AgreementUrlsConfig.kt` and `FirebaseConfig.kt`.
+
+## Getting Started
+
+### Configuration Setup
+
+Edit the `config_settings.yaml` in the `default_config` folder. It should contain data as follows:
+
+```yaml
+config_directory: '{path_to_config_folder}'
```

Comment:
For developer convenience, we should mention that the config folder can also be placed outside the repository. What do you think?

Reply:
Makes sense!
Added ✅